Firestore/core/test/unit/api/load_bundle_task_test.cc

```diff
@@ -380,36 +366,31 @@ TEST(LoadBundleTaskTest, ProgressObservesUntilError) {
 
   task.SetError();
   auto expected = Progress(2, 5);
-  expected.set_state(LoadBundleTaskState::Error);
+  expected.set_state(LoadBundleTaskState::kError);
   EXPECT_EQ(expected, queue.pop());
 
   EXPECT_TRUE(queue.empty());
 }
 
-TEST(LoadBundleTaskTest, ProgressObservesInitialError) {
-  LoadBundleTask task(CreateUserQueue());
+TEST_F(LoadBundleTaskTest, ProgressObservesInitialError) {
   BlockingQueue<LoadBundleTaskProgress> queue;
-  task.ObserveState(LoadBundleTaskState::InProgress,
+  task.ObserveState(LoadBundleTaskState::kInProgress,
                     [&](LoadBundleTaskProgress p) { queue.push(p); });
 
   task.SetError();
-  auto expected = LoadBundleTaskProgress();
-  expected.set_state(LoadBundleTaskState::Error);
-  EXPECT_EQ(expected, queue.pop());
 
+  EXPECT_EQ(ErrorProgress(), queue.pop());
   EXPECT_TRUE(queue.empty());
 }
 
-TEST(LoadBundleTaskTest, NoObserversAlsoWork) {
-  LoadBundleTask task(CreateUserQueue());
-
+TEST_F(LoadBundleTaskTest, NoObserversAlsoWork) {
   // No way to observe, simple checking no crashing happens.
-  task.UpdateProgress(InitialProgress());
-  task.UpdateProgress(Progress(2, 5));
-  task.SetSuccess(SuccessProgress());
-  task.SetError();
+  EXPECT_NO_THROW(task.UpdateProgress(InitialProgress());
```

Comment:
Looking at Googletest docs, maybe `EXPECT_NO_FATAL_FAILURE` is slightly more appropriate (since we don't throw, strictly speaking). Also, for readability, can you wrap the code in curly braces?
```cpp
EXPECT_NO_FATAL_FAILURE({
  task.UpdateProgress(InitialProgress());
  task.UpdateProgress(Progress(2, 5));
  // ...
});

Reply:
Done.